screwdriver/screwdriver_test.go

```diff
@@ -0,0 +1,268 @@
+package screwdriver
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var testDir string = "./testdata"
+
+func validateHeader(tb testing.TB, key, value string, r *http.Request) {
+	tb.Helper()
+	headers, ok := r.Header[key]
+
+	assert.True(tb, ok, fmt.Sprintf("No %s header sent in Screwdriver request", key))
+
+	header := headers[0]
+	assert.Equal(tb, value, header)
+}
+
+func TestNew(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		testToken := "token"
+		testJWT := "jwt"
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			wantAcceptMIMEType := "application/json"
+			validateHeader(t, "Accept", wantAcceptMIMEType, r)
+			token := r.URL.Query().Get("api_token")
+			assert.Equal(t, token, testToken)
+
+			w.WriteHeader(200)
+			w.Header().Set("Content-Type", "application/json")
+
+			testBody := fmt.Sprintf(`{"token": "%s"}`, testJWT)
+			fmt.Fprintln(w, testBody)
+		}))
+
+		gotAPI, err := New(server.URL, testToken)
+		api, ok := gotAPI.(*sdAPI)
+		assert.True(t, ok)
+		assert.Nil(t, err)
+		assert.Equal(t, testToken, api.UserToken)
+		assert.Equal(t, server.URL, api.APIURL)
+		assert.Equal(t, testJWT, api.SDJWT)
+		assert.Equal(t, testJWT, api.JWT())
+	})
+
+	t.Run("failure by invalid JSON", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(200)
+			w.Header().Set("Content-Type", "application/json")
+
+			testBody := fmt.Sprint(`{`)
+			fmt.Fprintln(w, testBody)
+		}))
+
+		testToken := "token"
+		_, err := New(server.URL, testToken)
+		assert.NotNil(t, err)
```

Comment:
Can we check the error is caused by `invalid JSON`?
I think just using `assert.NotNil` can't specify the error type.

Reply:
I fixed it.